packages/@aws-cdk/aws-lambda-event-sources/test/test.kafka.ts

```diff
@@ -0,0 +1,264 @@
+
+export = {
+  'msk': {
+    'default'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const fn = new TestFunction(stack, 'Fn');
+      const cluster = msk.Cluster.fromClusterArn(stack, 'Cluster', 'some-arn');
+      const kafkaTopic = 'some-topic';
+      const secret = new Secret(stack, 'Secret', { secretName: 'AmazonMSK_KafkaSecret' });
+
+      // WHEN
+      fn.addEventSource(new sources.ManagedKafkaEventSource(
+        {
+          cluster: cluster,
+          topic: kafkaTopic,
+          secret: secret,
+          startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+        }));
+
+      // THEN
+      expect(stack).to(haveResource('AWS::IAM::Policy', {
+        PolicyDocument: {
+          Statement: [
+            {
+              Action: [
+                'secretsmanager:GetSecretValue',
+                'secretsmanager:DescribeSecret',
+              ],
+              Effect: 'Allow',
+              Resource: {
+                Ref: 'SecretA720EF05',
+              },
+            },
+            {
+              Action: [
+                'kafka:DescribeCluster',
+                'kafka:GetBootstrapBrokers',
+                'kafka:ListScramSecrets',
+              ],
+              Effect: 'Allow',
+              Resource: cluster.clusterArn,
+            },
+          ],
+          Version: '2012-10-17',
+        },
+        PolicyName: 'FnServiceRoleDefaultPolicyC6A839BF',
+        Roles: [
+          {
+            Ref: 'FnServiceRoleB9001A96',
+          },
+        ],
+      }));
+
+      expect(stack).to(haveResource('AWS::Lambda::EventSourceMapping', {
+        EventSourceArn: cluster.clusterArn,
+        FunctionName: {
+          Ref: 'Fn9270CBC0',
+        },
+        BatchSize: 100,
+        StartingPosition: 'TRIM_HORIZON',
+        Topics: [
+          kafkaTopic,
+        ],
+        SourceAccessConfigurations: [
+          {
+            Type: 'SASL_SCRAM_512_AUTH',
+            URI: {
+              Ref: 'SecretA720EF05',
+            },
+          },
+        ],
+      }));
+
+      test.done();
+    },
+  },
+
+  'self managed kafka': {
+    'default'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+      const fn = new TestFunction(stack, 'Fn');
+      const kafkaTopic = 'some-topic';
+      const secret = new Secret(stack, 'Secret', { secretName: 'AmazonMSK_KafkaSecret' });
+      const bootstrapServers = ['kafka-broker:9092'];
+      const sg = SecurityGroup.fromSecurityGroupId(stack, 'SecurityGroup', 'sg-0123456789');
+      const vpc = new Vpc(stack, 'Vpc');
+
+      // WHEN
+      fn.addEventSource(new sources.SelfManagedKafkaEventSource(
+        {
+          bootstrapServers: bootstrapServers,
+          topic: kafkaTopic,
+          secret: secret,
+          startingPosition: lambda.StartingPosition.TRIM_HORIZON,
+          vpc: vpc,
+          vpcSubnets: { subnetType: SubnetType.PRIVATE },
+          securityGroup: sg,
```

Comment:
The 'default' test case is meant to test the output when only the required parameters are set.
Further test cases later add the optional properties and assert the appropriate behaviour is achieved.

vpc, sec groups and subnets are optional properties

Reply:
Applied changes.